minio_storage/storage.py

```diff
@@ -39,12 +41,12 @@ def __init__(
         bucket_name: str,
         *,
         base_url: T.Optional[str] = None,
-        file_class: T.Optional[T.Type[File]] = None,
+        file_class: T.Optional[T.Type[MinioStorageFile]] = None,
         auto_create_bucket: bool = False,
         presign_urls: bool = False,
         auto_create_policy: bool = False,
         policy_type: T.Optional[Policy] = None,
-        object_metadata: T.Optional[T.Dict[str, str]] = None,
+        object_metadata: T.Optional[ObjectMetadataType] = None,
```

Comment:
The real problem here is that Minio's annotations say that object metadata must be a `dict`, when really they ought to accept the more permissive `collections.abc.Mapping`, as the `Mapping` is covariant (i.e. if it requires a `dict[str, str | list[str]]`, we can pass a `dict[str, str]` and it will be happy.

To avoid imposing this same bug on our users, we correctly declare this parameter to be a `Mapping`, then just `type: ignore[arg-type]` the incompatibility when calling Minio APIs.

Another benefit of this change is to broaden what `object_metadata` is allowed, so users are now free to pass more sophisticated metadata structures (with values of lists or tuples).